deps/v8/include/v8.h

```diff
@@ -5381,6 +5381,17 @@ class V8_EXPORT Isolate {
   static Isolate* GetCurrent();
 
   /**
+   * Custom callback used by embedders to help V8 determine if it should abort
+   * when it throws and no internal handler can catch the exception.
+   * If FLAG_abort_on_uncaught_exception is true, then V8 will abort if either:
+   * - no custom callback is set.
+   * - the custom callback set returns true.
+   * Otherwise it won't abort.
+   */
+  typedef bool (*abort_on_uncaught_exception_t)(Isolate*);
```

Comment:
I'd combine it like this:

```c++
if (uncaught_exception_callback_ != nullptr &&
    PredictExceptionCatcher() != CAUGHT_BY_JAVASCRIPT) {
  uncaught_exception_callback_(reinterpret_cast<v8::Isolate*>(this));
} else if (FLAG_abort_on_uncaught_exception) {
  // ...
  base::OS::Abort();
}
```

In a perfect world `--abort_on_uncaught_exception` wouldn't exist in V8 and just be handled by the embedder.